packages/@aws-cdk/aws-appsync/README.md

```diff
@@ -285,6 +285,41 @@ ds.createResolver({
 });
 ```
 
+## Custom Domain Names
+
+For many use cases you may want to associate a custom domain name with your
+GraphQL API. This can be done during the API creation.
+
+```ts
+import { Certificate } from '@aws-cdk/aws-certificatemanager';
+
+declare const certificate: Certificate;
```

Comment:
Will this work? You're not initializing it to anything.

Reply:
Will this work? no. Can it compile this way? yes. This is something we are using in our examples for necessary properties that are not essential to the overall example. We don't want to clutter our examples with a `new lambda.Function(this, 'myfn', {...});` each time we need one, so it is simpler to write `declare const fn: lambda.Function` and leave it to the user to provide the exact function they need if they want to use the example.

That being said, I feel like providing a real `certificate` is necessary to this example, since it is part of the `domainName` property we are demonstrating. So I do think in this situation we should provide a real certificate.

Reply:
So I did not invent this pattern I modeled most of the work from the rest api pattern [here](https://docs.aws.amazon.com/cdk/api/v1/docs/aws-apigateway-readme.html#custom-domains)